draft-ietf-httpbis-priority.md

```diff
@@ -108,101 +114,100 @@ The term control stream is used to describe the HTTP/2 stream with identifier
+RFC 7540 describes denial-of-service considerations for implementations. On
 2019-08-13 Netflix issued an advisory notice about the discovery of several
-resource exhaustion vectors affecting multiple HTTP/2 implementations. One
-attack, [CVE-2019-9513] aka "Resource Loop", is based on manipulation of the
-priority tree.
-
-The HTTP/2 scheme depends on in-order delivery of signals, leading to challenges
-in porting the scheme to protocols that do not provide global ordering. For
-example, the scheme cannot be used in HTTP/3 {{HTTP3}} without
-changing the signal and its processing.
-
-Considering the problems with deployment and adaptability to HTTP/3, retaining
-the HTTP/2 priority scheme increases the complexity of the entire system without
-any evidence that the value it provides offsets that complexity. In fact,
-multiple experiments from independent research have shown that simpler schemes
-can reach at least equivalent performance characteristics compared to the more
-complex HTTP/2 setups seen in practice, at least for the web use case.
-
-## Disabling HTTP/2 Priorities {#disabling}
-
-The problems and insights set out above are motivation for allowing endpoints to
-opt out of using the HTTP/2 priority scheme, in favor of using an alternative
-such as the scheme defined in this specification. The
-SETTINGS_DEPRECATE_HTTP2_PRIORITIES setting described below enables endpoints to
-understand their peer's intention. The value of the parameter MUST
-be 0 or 1. Any value other than 0 or 1 MUST be treated as a connection error
-(see {{Section 5.4.1 of HTTP2}}) of type PROTOCOL_ERROR.
+resource exhaustion vectors affecting multiple RFC 7540 implementations. One
+attack, [CVE-2019-9513] aka "Resource Loop", is based on using priority signals
+to manipulate the server's stored prioritization state.
+
+HTTP/2 priority signals depend on in-order delivery, leading to challenges in
+porting the approach to protocols that do not provide global ordering. For
+example, it could not be supported in HTTP/3 {{HTTP3}} without changing the
+signals and their processing.
+
+Considering the problems with the deployment of RFC 7540 stream priority, and
+the difficulties in adapting it to HTTP/3, continuing to base prioritization on
+this mechanism risks increasing the complexity of systems. Multiple experiments
+from independent research have shown that simpler schemes can reach at least
+equivalent performance characteristics compared to the more complex RFC 7540
+setups seen in practice, at least for the web use case.
+
+## Disabling RFC 7540 Priorities {#disabling}
+
+The problems and insights set out above provided the motivation for deprecating
+RFC 7540 stream priority (see {{Section 5.3 of RFC7540}}).
+
+The SETTINGS_DEPRECATE_RFC7540_PRIORITIES setting is defined by this document in
+order to allow endpoints to explicitly opt out of using HTTP/2 priority signals
+(see {{Section 5.3.2 of HTTP2}}). Endpoints are expected to use an alternative,
```

Comment:
Here, it's somewhat unclear who sends this setting and who acts upon it. You use "endpoint" for both client and server interchangeably (which is correct, but somewhat confusing). 

As in: does the client use the setting to indicate it won't use the signals? or just the server to indicate it won't? Later text first seems to indicate it's only the server sending it, but at the end, it's mentioned the client can also send... imo, this should be clarified up-front: who can send it and what are the semantics from both sides (e.g., if client send it, does it promise not to send RFC 7540 signals?)

Reply:
This question is tangential to the editorial change at hand. Personally I don't see a problem with the text as it is. But if you feel strongly a new issue can be opened.